docs/wip/how-to-rollback.md

```diff
@@ -8,5 +54,57 @@
 - Caching? How much time it takes to invalidate caching
 
 
+# Expected Behavior
+When the rollback is performed with a `2xx` status response, it will trigger a new build of your store from the branch `backup-branch`.
+Because of that, the time to replace the current store can take about 10 minutes, depending on the size of your store.
+
+All the subsequent production builds of your store will be based on that commit from the `backup-branch`.
+Once the bug is fixed, to go back to the `main` version of your store, you need to perform the request:
```

Comment:
Isn't making this request after the bug is fixed in the main branch redundant since the merge will update the target branch, as mentioned below:

```
It will also allow new builds from the `main` branch everytime a merge happens.
```

And some questions:
- what kind of authorization is done when calling this API? specifically, what happens if I provide an io token from account X to update the target branch/commit of account Y?
- If the target branch is `backup-branch` and someone merges another branch into it or pushes a commit, will that trigger a build? (I think the current behavior will change depending on whether the incoming commit belongs to a PR) If so, will the master route (i.e marinbrasil.vtex.app) point to the new deploy preview or would the user need to call the API again after pushing to `backup-branch`

Reply:
I'm just checking about when the bug is fixed in the default branch, but I'm confident you're right

I'll add the details of your questions to the documentation. But I'll leave it also here for reference
- When the authorization fails, the response status is `4xx`. The rollback will take place when the response status is `201`. The token comes from `vtex local token` and the autorization is the one with VTEX ID, checking whether the token is capable of accessing the `workspace-read-write` resource. I've found no part of the code that protects from cross-account rollback. I'm adding it to our backlog
- There is no logic for that in place. There can be as many pushes to the non-default branch where the rollback happened, and it will not change

I'm just checking some details, I'll rework this part of the documentation